### ISSUE TITLE
Cloud storage (FTP) doesn't work when password contains "@" 

### ISSUE THREAD

**Issue by mwoz123** [koreader/koreader#3641]
* KOReader version: todays git
* Device: pc emulator , android

#### Issue
"Cannot fetch list of folder contents. please check your configuration..."

Edit: password containing "@" - can connect.

#### Steps to reproduce
My configuration:
EDIT: Removed image. 
Config that doesn't work below: https://github.com/koreader/koreader/issues/3641#issuecomment-362921761

~~Doesn't seem to work with nor without password.~~

##### In console 
can't see usefull info :

01/29/18-22:43:50 DEBUG creating coroutine for menu select
PING 10.0.0.1 (10.0.0.1) 56(84) bytes of data.
64 bytes from 10.0.0.1: icmp_seq=1 ttl=64 time=0.392 ms

--- 10.0.0.1 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.392/0.392/0.392/0.000 ms
01/29/18-22:44:21 DEBUG show widget unknown
01/29/18-22:44:21 DEBUG blitFrom 7 670 0 0 48 48
01/29/18-22:44:21 DEBUG ImageWidget: _render'ing resources/info-i.png
01/29/18-22:44:21 DEBUG blitFrom 55 330 0 0 60 60
01/29/18-22:44:21 DEBUG refresh on physical rectangle -1 90 529 51
01/29/18-22:44:21 DEBUG refresh on physical rectangle 47 313 445 94
01/29/18-22:44:21 DEBUG BackgroundRunner: _execute() @  1517262261
01/29/18-22:44:21 DEBUG BackgroundRunnerWidget: start running @  1517262261

**Comment by Frenzie** [comment]
Works for me with `ftp://ftp.heanet.ie`.

Pinging @robert00s

**Comment by robert00s** [comment]
I cannot check without a password.

**Comment by mwoz123** [comment]
Try our local:
ftp: mkwk018.cba.pl

![2018-01-30 22-04-44](https://user-images.githubusercontent.com/4487025/35591266-a4fa55b2-0609-11e8-89dd-df913b4a9a55.png)


The same error

**Comment by mwoz123** [comment]
Maybe other way what free ftp servers work with this plugin? 

I can't see registration on 'heanet.ie', (only for Ireland students?)

**Comment by Frenzie** [comment]
It's a public FTP server. Its main use is for Debian.

Other FTP servers that (probably) work. I didn't try all of them, just a random sampling:

ftp.altlinux.org
ftp.archlinux.org
ftp.debian.org
ftp.exim.org
ftp.freebsd.org
ftp.frugalware.org
ftp.gimp.org
ftp.gnome.org
ftp.gnu.org
ftp.gnupg.org
ftp.ietf.org
ftp.isc.org
ftp.kde.org
ftp.netbsd.org
ftp.openbsd.org
ftp.openssl.org
ftp.opensuse.org
ftp.perl.org
ftp-pingback
ftp.porcupine.org
ftp.postgresql.org
ftp.proftpd.org
ftp.ripe.net
ftp.samba.org
ftp.slackware.com
ftp.vim.org
ftp.xemacs.org

**Comment by mwoz123** [comment]
But they doesn't seem to be personal/private data hosting ftps..

And I would like to use koreader ftp plugin for personal books sending, not to download Debian or open source os/app :p

3 of 3 personal free ftps that I checked wasn't able to connect:(

Any personal  ftp hosting that works with this plugin?;)

**Comment by Frenzie** [comment]
Your example also works for me (except that in your screenshot it doesn't have ftp:// in the address).

![screenshot_2018-02-01_15-06-23](https://user-images.githubusercontent.com/202757/35682591-7cfb9950-0761-11e8-8eb4-1aecb080a065.png)

![screenshot_2018-02-01_15-05-27](https://user-images.githubusercontent.com/202757/35682575-731763b0-0761-11e8-916b-c167bbb0ad38.png)

![screenshot_2018-02-01_15-05-35](https://user-images.githubusercontent.com/202757/35682576-7331e5aa-0761-11e8-9a86-a3a7d3a5c154.png)

**Comment by mwoz123** [comment]
I finally discovered what was the reason.

**Problem is invalid encoding "@" sign** I had in my password. Then it cannot connect.

I changed password to containg "@", current (fail) connection details :

![2018-02-04 17-53-30](https://user-images.githubusercontent.com/4487025/35779913-c057cf14-09d4-11e8-9793-64b5da9c1b61.png)

Now it gives the same result I've: 
![2018-02-04 17-54-28](https://user-images.githubusercontent.com/4487025/35779916-c8d06354-09d4-11e8-9bef-8bf4090b58c8.png)
